### ISSUE TITLE
Training/Validation augmentation/batch/shuffle strategy

### ISSUE THREAD

**Issue by asmagen** [qubvel/segmentation_models#351]
In the multi-class tutorial here you provide the following strategy for train/valid loading:
```
# Dataset for train images
train_dataset = Dataset(
    x_train_dir, 
    y_train_dir, 
    classes=CLASSES, 
    augmentation=get_training_augmentation(),
    preprocessing=get_preprocessing(preprocess_input),
)

# Dataset for validation images
valid_dataset = Dataset(
    x_valid_dir, 
    y_valid_dir, 
    classes=CLASSES, 
    augmentation=get_validation_augmentation(),
    preprocessing=get_preprocessing(preprocess_input),
)

train_dataloader = Dataloder(train_dataset, batch_size=BATCH_SIZE, shuffle=True)
valid_dataloader = Dataloder(valid_dataset, batch_size=1, shuffle=False)
```
What's the reason to perform augmentations on the validation set and not only on the training set? and what's the role of the shuffling and greater than 1 batch size (depending on the parameter, while it doesn't in the validation) only in the training set?

**Comment by JordanMakesMaps** [comment]
Whoops! I misread your question. I thought you were asking why are images **and** masks augmented. Derp. Well, either way, you can still augment the validation set as well, but you're right: usually you wouldn't augment the validation set.

But notice what `get_validation_augmentation()` is actually doing. It's just padding the image, so it's not really augmenting but rather its performing a form of pre-processing to match what is done on the training set.
